src/map/ai/states/item_state.h

```diff
@@ -0,0 +1,58 @@
+/*
+ ===========================================================================
+
+ Copyright (c) 2010-2015 Darkstar Dev Teams
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see http://www.gnu.org/licenses/
+
+ This file is part of DarkStar-server source code.
+
+ ===========================================================================
+ */
+
+#ifndef _CITEM_STATE_H
+#define _CITEM_STATE_H
+
+#include "state.h"
+#include "../../items/item_usable.h"
+
+struct action_t;
+
+class CItemState : public CState
+{
+public:
+    CItemState(CBattleEntity* PEntity, uint16 targid, uint8 loc, uint8 slotid);
+    virtual bool Update(time_point tick) override;
+    virtual void Cleanup(time_point tick) override;
+    virtual bool CanChangeState() override;
+    virtual bool CanFollowPath() override { return false; }
+
+    virtual void TryInterrupt(CBattleEntity* PAttacker) override;
+
+    CItemUsable* GetItem();
+
+    void InterruptItem(action_t& action);
+    void FinishItem(action_t& action);
+
+protected:
+    bool HasMoved();
+
+    CBattleEntity* const m_PEntity;
```

Comment:
switch this to a CCharEntity* and you won't have to static cast it to a CharEntity everywhere